### PR TITLE
Regional compilation support

### PR DESCRIPTION
**Pull request by Kacper-Pietkun**
Add support for regional compilation. It is turned on by default, but can be turned off with `VLLM_REGIONAL_COMPILATION` env variable. It works only for torch.compile execution mode. It significantly speeds up warmup time and slightly increases throughput.

### CODE REVIEW

File: vllm/worker/hpu_model_runner.py [hunk 1]
```diff
@@ -1538,7 +1558,7 @@ def warmup_model(self, kv_caches: List[torch.Tensor]) -> None:
         self.bucketing_ctx.generate_decode_buckets(max_blocks)
 
         if not htorch.utils.internal.is_lazy() and not self.enforce_eager:
-            cache_size_limit = len(self.bucketing_ctx.prompt_buckets) + len(
+            cache_size_limit = 3 * len(self.bucketing_ctx.prompt_buckets) + 3 * len(
                 self.bucketing_ctx.decode_buckets) + 1
```
Comment (anko-intel) [inline vllm/worker/hpu_model_runner.py:1562]: MAybe:
```suggestion
            cache_size_limit = 1 + 3  *
             (len(self.bucketing_ctx.prompt_buckets) + len(self.bucketing_ctx.decode_buckets))
```

**Review by anko-intel — APPROVED** [review]
LGTM

File: vllm/worker/hpu_model_runner.py (hunk 1 above)
Comment (Kacper-Pietkun) [reply vllm/worker/hpu_model_runner.py]: done

**Review by jczaja — APPROVED** [review]
LGTM

**Comment by jczaja** [comment]
@rsshaik1 This is PR introducing torch compile regional compilation. It would be good know if this PR fixes the problem you worked at #528.

**Review by madamczykhabana — APPROVED** [review]
LGTM

**Review by RafLit — APPROVED** [review]
